server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java

```diff
@@ -193,13 +195,17 @@ private static Tuple<String, String> getValue(Map<String, String> attributes, Li
     private Map<String, String> getValidatedRepositoryNames(DiscoveryNode node) {
         Set<Tuple<String, String>> repositoryNames = new HashSet<>();
         if (containsKey(node.getAttributes(), REMOTE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEYS)
-            || containsKey(node.getAttributes(), REMOTE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEYS)) {
+            && containsKey(node.getAttributes(), List.of(REMOTE_STORE_SEGMENTS_ONLY_ATTRIBUTE_KEY))
```

Comment:
I don't love that we need to introduce a new attribute to do what you're trying to do... its weird config to me to say "here's my segment repo and here's another setting bc I want only that".  With that said I understand why you're doing it.  If we decouple these we push that validation onto an operator and thats post node join...

The only alternative I can think of is a separate "strict" setting which is essentially the same thing as this.